pilot/pkg/model/service.go

```diff
@@ -349,6 +349,9 @@ type IstioEndpoint struct {
 
 	// TLSMode endpoint is injected with istio sidecar and ready to configure Istio mTLS
 	TLSMode string
+
+	// EffectiveTlsMode is the result of combined TLSMode and peer authnetication policy.
+	EffectiveTlsMode string
```

Comment:
It is confusing, can we remove this and use TLSMode instead  ?

Reply:
indeed it's a bit confusing. i initially want to avoid adding a field in this struct as well. but

1. If we just overwrite the `TLSMode` with `disabled` when a policy configures it as `mtls: DISABLE`, then it can never recover back later when we change policy to `mtls: STRICT`. because the information that this endpoint has been deployed with a sidecar is lost during override.
1. I can't just do in place temporary variable when passing to `BuildLbEndpointMetadata` either. this is because
`BuildLbEndpointMetadata` has two caller usage. one is in `[toEnvoyEndpoint](https://github.com/istio/istio/blob/5e6b923525cc6ee8f99dce7971b0d2e8547ccf66/pilot/pkg/proxy/envoy/v2/eds.go#L177)` the other is in `buildEnvoyLbEndpoint`.

Reply:
@howardjohn anser for the same question you have...

Reply:
yeah it didn't really answer for me, why can't we just generate at the 2 call sites?